File: contests/2016/bronze/daisy_chains.cpp
#include <bits/stdc++.h>

using namespace std;

typedef int64_t ll;

void dbg(const vector<int>& v){
    for(int x:v)
        cout<<x<<' ';
    cout<<'\n';
}


int main(){
    ios_base::sync_with_stdio(false); cin.tie(NULL);
    int n,s; cin>>n;
    vector<int> v(n);

    for(int &x:v)
        cin>>x;

    int res=0;
    double avg;

    for(int i=0;i<n;i++){
        for(int j=i;j<n;j++){
            s=0;
            for(int k=i;k<=j;k++)
                s+=v[k];
            avg=(double)s/(j-i+1);
            for(int k=i;k<=j;k++){
                if(v[k]==avg){
                    ++res;
                    break;
                }
            }
        }
    }
    cout<<res;

    return 0;
}

File: contests/2016/bronze/daisy_chains.py
nums=[1,1,2,3]
n=len(nums)

for i in range(n):
    for j in range(i,n):
        s=0
        for k in range(i,j+1):
            s+=nums[k]
        avg=s//(j-i+1)
        for k in range(i,j+1):
            if(nums[k]==avg):
                res+=1


print(res)

